### PR TITLE
Activity details loading

### PR DESCRIPTION
**Pull request by greg-schrammel**
Fixes BX-####
Figma link (if any):

## What changed (plus any additional context for devs)

- activity details sheet insta shows on clicking an activity (before was waiting for loading), now it shows with the data we got and updates as more data comes
- removed `testnetMode` from the consolidatedTransactionsQueryKey as it was not used in the query itself and I didn't find a reason to have it in the query key lmk if Im mistaken
- refactored a lil bit the `useTransaction` hook, aggregated into a single useQuery for either supported or unsupported BE networks and added some more info we could find when fetching the tx data from the provider

### CODE REVIEW

**Review by derHowie — APPROVED** [review]

File: src/core/resources/transactions/transaction.ts [hunk 1]
```diff
@@ -52,6 +51,14 @@ export const fetchTransaction = async ({
   currency: SupportedCurrencyKey;
   chainId: ChainId;
 }) => {
+  if (!SUPPORTED_CHAIN_IDS.includes(chainId)) {
```
Comment (estebanmino) [inline src/core/resources/transactions/transaction.ts:54]: did you find any case where we pass unsopperted chainids?

all of the instances are using chains coming from useBackendSupportedChains hook like this https://github.com/rainbow-me/browser-extension/blob/master/src/entries/popup/hooks/useTransactionListForPendingTxs.ts#L32 or from that constant https://github.com/rainbow-me/browser-extension/blob/master/src/entries/popup/pages/home/Approvals/Approvals.tsx#L342

File: src/core/resources/transactions/transaction.ts [hunk 2]
```diff
@@ -83,133 +94,93 @@ export const fetchTransaction = async ({
...
+
+      gasUsed: receipt.gasUsed.toString(),
+
+      feeType: !block.baseFeePerGas ? 'legacy' : 'eip-1559',
+      fee: receipt.cumulativeGasUsed.mul(receipt.effectiveGasPrice).toString(),
+      baseFee: block.baseFeePerGas?.toString(),
+    };
+  }
+
+  const type = await guessTransactionType(provider, transaction);
```
Comment (estebanmino) [inline src/core/resources/transactions/transaction.ts:173]: why await?

File: src/core/resources/transactions/transaction.ts [hunk 3]
```diff
@@ -218,32 +189,53 @@ export const useTransaction = ({
...
+        if (tx) return tx;
+      }
+
+      // try to find in custom network transactions store
+      const { getCustomNetworkTransactions } =
+        customNetworkTransactionsStore.getState();
+      return getCustomNetworkTransactions({ address }).find(
+        (tx) => tx.hash === hash && tx.chainId === chainId,
+      );
+    },
```
Comment (estebanmino) [inline src/core/resources/transactions/transaction.ts:234]: i don't follow why this is mixing txs supported by backend and the ones from `customNetworkTransactionsStore`, there are separate stores and we're using them separately

File: src/core/resources/transactions/transaction.ts (hunk 1 above)
Comment (greg-schrammel) [reply src/core/resources/transactions/transaction.ts:54]: I removed the `useBackendSupportedChains` hook didn't understand why we had two hooks to fetch txs, replaced with a single `useTransaction` (which from outside this file works the same) to handle the transaction metadata no matter where that data is coming from

File: src/core/resources/transactions/transaction.ts (hunk 2 above)
Comment (greg-schrammel) [reply src/core/resources/transactions/transaction.ts:173]: Im fetching data from the provider to guess the type
like `transaction.to` has code (`provider.getCode`) means it's probably a transaction interacting with a contract

Comment (greg-schrammel) [reply src/core/resources/transactions/transaction.ts:173]: hmm idk whats the expected behavior on smart wallets

File: src/core/resources/transactions/transaction.ts (hunk 1 above)
Comment (estebanmino) [reply src/core/resources/transactions/transaction.ts:54]: okok it makes sense

**Review by estebanmino — APPROVED** [review]
lgtm 🚀

**Pull request merged**